how-to-docs/amazon-resources.md

```diff
@@ -0,0 +1,55 @@
+# How to get Amazon Resources in the UK Cluster
+
+## Workflow
+1. Pull stacks-hod-platform repo
+2. Create/request credentials and secrets
+3. Configure templates in feature branch
+4. Create a pull request, and an _Amazon Resource Request_ in the hub
+5. After review, your pull request will be merged and your resources created
+
+### Pull the stacks-hod-platform repo
+```
+git clone ssh://git@gitlab.digital.homeoffice.gov.uk:2222/Devops/stacks-hod-platform.git
+cd stacks-hod-platformhttps://gitlab.digital.homeoffice.gov.uk/Devops/stacks-hod-platform/
+git checkout -b myexample-s3
+```
+
+### Configuring stacks
+
+Most new provisioning requests are similar, copy an existing template in stacks/templates/hod-dsp-[env]
+### For RDS
+
+1. Copy a similar existing template based on the db type (postgres/mysql or other). Change the lines at the top defining the resource name, this example satisfies the request of the name egteam in the prod environment. Other changes may be required such as TCP Port etc.
+
+```
+{% set service_name='egteam' %}
+{% set service_env='prod' %}
+{% set dns_name=service_name+'-'+service_env %}
+# the resource name constructed like: {{ service_name }}-{{ service_env }}-rds
+```
+This will be visible in the deployed AWS resource as DBSubnetGroupName": "egteam-prod-rds-..."
+
+2. Change the instance type and database size according to the request 
+```
+AllocatedStorage: 100
+DBInstanceClass: db.m3.large
+```
+3. Name the password variable appropriately and ensure service not tagged public 
+```
+MasterUserPassword: {{ egteam_prod_rds_root_password }}
+PubliclyAccessible: false
+```
+
+4. create an amazon resource request in the platform hub referring to your request, and your namespace. Your aws credentials will be added to your namespace as a base64 encoded secret.
+
+### For S3 buckets
+1. Create an issue on the bau board requesting an iam user for programmatic access with kms enabled. Your kms key will be sent to you.
+2. Configure stacks and create a merge request. For an S3 template, replace the KMS key at the top, with the KMS key you just obtained.
+```
+{% set service_name = 'myexample' %}
+{% set kms_id = '58855abe-2f87-4e61-9ac1-ef29b6254438' %}
+{% set aws_user = service_name.upper() + '_' + env.upper() + '_S3' %}
+{% set service_bucket = service_name + '-' + env + region %}
```

Comment:
As mentioned above we should probably have a terraform module to do all this creation for you, including the KMS key, S3 bucket, IAM policies etc. And then this code snippet here should explain how to use that module and what variables you pass to it.